README.md

```diff
@@ -1,48 +1,207 @@
+confusing errors.
+
+Imagine the phone number above being saved to the database in an application
+that makes automatic calls using the Twilio API. When your system tries to call
+this number, there will be an error because no such phone number exists, which
+means you need to have an entire branch of code dedicated to handling *just*
+that edge case.
+
+It would be much easier if you never have bad data in the first place, so you
+can focus on handling edge cases that are truly unpredictable.
+
+That's where validations come in.
+
+
+# Basic Usage
+
+For more examples of basic validation usage, see the Rails Guide for [Active
+Record Validations][ar_validations]. Take a few minutes to browse the helpers
+listed in Section 2.
+
+```ruby
+class Person < ActiveRecord::Base
+  validates :name, presence: true
+end
+
+Person.create(name: "John Doe").valid? # => true
+Person.create(name: nil).valid? # => false
+```
+
+[ar_validations]: http://guides.rubyonrails.org/active_record_validations.html
+
+`#validates()` is our Swiss Army knife for validations. It takes two arguments:
+the first is the name of the attribute we want to validate, and the second is a
+hash of options that will include the details of how to validate it.
+
+In this example, the options hash is `{ presence: true }`, which implements the
+most basic form of validation, preventing the object from being saved if its
+`name` attribute is empty.
+
+# Lifecycle Timing
+
+Before proceeding, keep the answer to this question in mind:
+
+**What is the difference between `#new()` and `#create()`?**
+
+If you've forgotten, `#new()` instantiates a new ActiveRecord model *without*
+saving it to the database, whereas `#create()` immediately attempts to save it,
+as if you had called `#new()` and then `#save()`.
+
+**Database activity triggers validation**. An ActiveRecord model instantiated
+with `#new()` will *not* be validated, because no attempt to write to the
+database was made.
+
+The only way to trigger validation without touching the database is to call the
+`#valid?()` method.
+
+For a full list of methods that trigger validation, see [Section
+4][ar_callbacks_4] of the Rails Guide for Active Record Callbacks. Don't worry
+about the rest of the information in that guide just yet; we'll go into
+callbacks later!
+
+[ar_callbacks_4]: http://guides.rubyonrails.org/active_record_callbacks.html#running-callbacks
+
+# Validation Failure
+
+Here it is, the moment of truth. What can we do when a record fails validation?
+
+## How can you tell when a record fails validation?
+
+**Pay attention to return values!**
+
+By default, ActiveRecord does not raise an exception when validation fails.
+The database operation method will simply return `false` and decline to update
+the database.
+
+Every database method has a sister method with a `!` at the end which will raise
+an exception (`#create!()` instead of `#create()` and so on).
+
+And of course, you can always check manually with `#valid?()`.
```

Comment:
Also, I think it would be useful to see `#valid` used in the example below if it is mentioned here.